module1/homework/lengthSort/sort.cpp

```diff
@@ -1,11 +1,14 @@
 #include "sort.hpp"
 
-#include <functional>
+std::deque<std::string> lengthSort(std::forward_list<std::string>& forwardList) {
+    std::deque<std::string> dq;
+    auto it = dq.begin();
 
-std::deque<std::string> lengthSort(std::forward_list<std::string> mail) {
-    mail.sort(std::greater<std::string>());
-    mail.reverse();
-    std::deque<std::string> vec{mail.begin(), mail.end()};
-
-    return vec;
+    for (const auto& word : forwardList) {
+        for (it = dq.begin(); it != dq.end(); it++)
+            if (it->size() > word.size() || it->size() == word.size() && *it > word)
+                break;
+        dq.insert(it, word);
+    }
```

Comment:
what about using `sort` with a lambda instead of double for loop? :)

Reply:
I don't think so :)

Reply:
Why? It's STL exercise so we don't want to have raw loops :)
Use sort algorithm here.